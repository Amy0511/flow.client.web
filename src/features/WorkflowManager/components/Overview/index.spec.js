import React from "react";
import renderer from "react-test-renderer";
import { Overview } from "./index";

jest.mock("./CronJobModal", () => "CronJobModal");
jest.mock("@boomerang/boomerang-components", () => ({
  AlertModal: "AlertModal",
  Button: "Button",
  ConfirmModal: "ConfirmModal",
  Modal: "ModalWrapper",
  ModalFlow: "ModalFlow",
  SelectDropdown: "SelectDropdown",
  notify: "notify",
  Notification: "Notification",
  TextArea: "TextArea",
  TextInput: "TextInput",
  Toggle: "Toggle",
  Tooltip: "Tooltip"
}));

const mockfn = jest.fn();

const workflow = {
  data: {
    triggers: {
      scheduler: {
        enable: false,
        schedule: ""
      },
      webhook: {
        enable: false,
        token: ""
      }
    }
  }
};

const enabledWorkflow = {
  data: {
    triggers: {
      scheduler: {
        enable: true,
        schedule: ""
      },
      webhook: {
        enable: false,
        token: ""
      }
    }
  }
};
const teams = [
  {
    id: "1234",
    name: "Lucas' team",
    workflows: [
      {
        id: "5bd6fdf65a5df954ad47bdbe",
        name: "lucas-workflow-1",
        description: "blablabla",
        status: "published",
        icon: "utility",
        properties: [
          {
            defaultValue: "test",
            description: "Testing this property",
            key: "test.property",
            label: "Test Property",
            required: true,
            type: "select",
            validValues: ["test", "this"]
          },
          {
            defaultValue: "test",
            description: "Testing this propertys",
            key: "test.property1",
            label: "Test Property1",
            required: true,
            type: "text"
          }
        ]
      },
      {
        id: "456",
        name: "lucas-workflow-2",
        description: "blablabla",
        status: "published",
        icon: "utility"
      },
      {
        id: "789",
        name: "lucas-workflow-3",
        description: "blablabla",
        status: "draft",
        icon: "secure"
      }
    ]
  }
];

const props = {
  teams,
  workflow,
  workflowActions: {
    updateTriggersWebhook: mockfn,
    updateProperty: mockfn,
    updateTriggersScheduler: mockfn
  },
  teamsActions: {},
  setIsValidOverview: mockfn
};

describe("Overview --- Snapshot", () => {
  it("+++ renders correctly", () => {
    const renderedValue = renderer.create(<Overview {...props} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("Overview --- Shallow render", () => {
  let wrapper;

  wrapper = shallow(<Overview {...props} />);

  it("renders with correct container class", () => {
    expect(wrapper.find(".c-worklfow-overview").length).toEqual(1);
  });
});
