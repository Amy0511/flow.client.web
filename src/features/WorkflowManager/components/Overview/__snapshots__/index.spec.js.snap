// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview --- Snapshot +++ renders correctly 1`] = `
<div
  className="c-worklfow-overview"
>
  <div
    className="c-overview-card"
  >
    <h1
      className="s-general-info-title"
    >
      General
    </h1>
    <div
      className="bmrg--c-select-dropdown --bmrg-white"
      style={
        Object {
          "marginBottom": "2rem",
        }
      }
    >
      <div
        className="s-select-dropdown-label"
      >
        Team
      </div>
      <div
        className="Select has-value is-searchable Select--single"
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-2--value"
          >
            <div
              className="Select-value"
            >
              <span
                aria-selected="true"
                className="Select-value-label"
                id="react-select-2--value-item"
                role="option"
              >
                Lucas' team
              </span>
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-activedescendant="react-select-2--value"
                aria-expanded="false"
                aria-haspopup="false"
                aria-owns=""
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <div
              className="bmrg--b-select-default-arrow"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="bmrg--c-text-input"
    >
      <label
        className="bmrg--s-text-input-title --visible --bmrg-white"
      >
        Name
      </label>
      <div
        className="bmrg--b-text-input"
      >
        <input
          className="bmrg--b-text-input__input --bmrg-white"
          disabled={false}
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="bmrg--c-text-input"
    >
      <label
        className="bmrg--s-text-input-title --visible --bmrg-white"
      >
        Summary
      </label>
      <div
        className="bmrg--b-text-input"
      >
        <input
          className="bmrg--b-text-input__input --bmrg-white"
          disabled={false}
          name="shortDescription"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Summary"
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="bmrg--b-textarea"
    >
      <label
        className="bmrg--b-textarea__title --visible --bmrg-white"
      >
        Description
      </label>
      <textarea
        className="bmrg--b-textarea__text"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Description"
        required={false}
        value=""
      />
    </div>
    <h2
      className="s-workflow-icons-title"
    >
      Icon
    </h2>
    <ul
      className="b-workflow-icons"
    >
      <li>
        <img
          alt="docs icon"
          className="b-workflow-icons__icon"
          onClick={[Function]}
          src="document_32.svg"
        />
      </li>
      <li>
        <img
          alt="secure icon"
          className="b-workflow-icons__icon"
          onClick={[Function]}
          src="secure.svg"
        />
      </li>
      <li>
        <img
          alt="flow icon"
          className="b-workflow-icons__icon"
          onClick={[Function]}
          src="flow.svg"
        />
      </li>
      <li>
        <img
          alt="log icon"
          className="b-workflow-icons__icon"
          onClick={[Function]}
          src="log.svg"
        />
      </li>
      <li>
        <img
          alt="payment icon"
          className="b-workflow-icons__icon"
          onClick={[Function]}
          src="payment.svg"
        />
      </li>
      <li>
        <img
          alt="diagram icon"
          className="b-workflow-icons__icon"
          onClick={[Function]}
          src="diagram.svg"
        />
      </li>
      <li>
        <img
          alt="utility icon"
          className="b-workflow-icons__icon"
          onClick={[Function]}
          src="utility.svg"
        />
      </li>
      <li>
        <img
          alt="template icon"
          className="b-workflow-icons__icon"
          onClick={[Function]}
          src="template.svg"
        />
      </li>
    </ul>
  </div>
  <div
    className="c-overview-card"
  >
    <h1
      className="s-trigger-title"
    >
      Triggers
    </h1>
    <div
      className="c-webhook"
    >
      <form
        className="b-webhook"
      >
        <label
          className="b-webhook__title"
          id="toggle-webhook"
        >
          Enable Webhook
        </label>
        <div
          className="b-webhook__toggle --bmrg-white --red"
        >
          <div
            className="react-toggle"
            onClick={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="react-toggle-track"
            >
              <div
                className="react-toggle-track-check"
              />
              <div
                className="react-toggle-track-x"
              />
            </div>
            <div
              className="react-toggle-thumb"
            />
            <input
              aria-labelledby="toggle-webhook"
              checked={false}
              className="react-toggle-screenreader-only"
              name="webhook"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
            />
          </div>
        </div>
        <img
          alt="Toggle webhook"
          className="b-options__infoIcon"
          data-for="triggers-webhook-info"
          data-tip={true}
          src="info.svg"
        />
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
          id="triggers-webhook-info"
        >
          Enable workflow to be executed by a webhook
        </div>
      </form>
      
      <div
        className="c-scheduler"
      >
        <div
          className="b-schedule"
        >
          <label
            className="b-schedule__title"
            id="toggle-scheduler"
          >
            Enable Scheduler
          </label>
          <div
            className="b-schedule__toggle --bmrg-black --red"
          >
            <div
              className="react-toggle"
              onClick={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="react-toggle-track"
              >
                <div
                  className="react-toggle-track-check"
                />
                <div
                  className="react-toggle-track-x"
                />
              </div>
              <div
                className="react-toggle-thumb"
              />
              <input
                aria-labelledby="toggle-scheduler"
                checked={false}
                className="react-toggle-screenreader-only"
                name="schedule"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <img
            alt="Toggle scheduler"
            className="b-options__infoIcon"
            data-for="triggers-scheduler-info"
            data-tip={true}
            src="info.svg"
          />
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
            id="triggers-scheduler-info"
          >
            Enable workflow to be executed by a schedule
          </div>
        </div>
        <div
          className="b-schedule__cronMessage"
        />
        <div
          className="b-schedule__timezone"
        />
      </div>
    </div>
  </div>
  <div
    className="c-overview-card"
  >
    <h1
      className="s-trigger-title"
    >
      Options
    </h1>
    <div
      className="b-options"
    >
      <form
        className="b-persistence"
      >
        <label
          className="b-persistence__title"
          id="toggle-persistence-storage"
        >
          Enable Persistent Storage
        </label>
        <div
          className="b-persistence__toggle --bmrg-white --red"
        >
          <div
            className="react-toggle"
            onClick={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="react-toggle-track"
            >
              <div
                className="react-toggle-track-check"
              />
              <div
                className="react-toggle-track-x"
              />
            </div>
            <div
              className="react-toggle-thumb"
            />
            <input
              aria-labelledby="toggle-persistence-storage"
              className="react-toggle-screenreader-only"
              name="persistence"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
            />
          </div>
        </div>
        <img
          alt="Toggle persistence"
          className="b-options__infoIcon"
          data-for="options-persistence-info"
          data-tip={true}
          src="info.svg"
        />
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
          id="options-persistence-info"
        >
          Persist workflow data between executions
        </div>
      </form>
    </div>
  </div>
</div>
`;
